binding/dotnet/PvSpeaker/PvSpeaker.cs

```diff
@@ -0,0 +1,441 @@
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern void pv_speaker_delete(IntPtr handle);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern PvSpeakerStatus pv_speaker_start(IntPtr handle);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern PvSpeakerStatus pv_speaker_stop(IntPtr handle);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern PvSpeakerStatus pv_speaker_write(IntPtr handle, IntPtr pcm, int pcmLength, out int writtenLength);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern PvSpeakerStatus pv_speaker_flush(IntPtr handle, IntPtr pcm, int pcmLength, out int writtenLength);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern PvSpeakerStatus pv_speaker_write_to_file(IntPtr handle, IntPtr outputPath);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern char pv_speaker_get_is_started(IntPtr handle);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr pv_speaker_get_selected_device(IntPtr handle);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern PvSpeakerStatus pv_speaker_get_available_devices(out int deviceListLength, out IntPtr deviceList);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern void pv_speaker_free_available_devices(int deviceListLength, IntPtr deviceList);
+
+        [DllImport(LIBRARY, CallingConvention = CallingConvention.Cdecl)]
+        private static extern IntPtr pv_speaker_version();
+
+        /// <summary>
+        /// Factory method for creating instances of PvSpeaker.
+        /// </summary>
+        /// <param name="sampleRate">
+        /// The sample rate of the audio to be played.
+        /// </param>
+        /// <param name="bitsPerSample">
+        /// The number of bits per sample of the audio to be played.
+        /// </param>
+        /// <param name="bufferSizeSecs">
+        /// The size in seconds of the internal buffer used to buffer PCM data
+        /// - i.e. internal circular buffer will be of size `sampleRate` * `bufferSizeSecs`.
+        /// </param>
+        /// <param name="deviceIndex">
+        /// The index of the audio device to play audio from. A value of (-1) will use the default audio device.
+        /// </param>
+        /// <returns>An instance of PvSpeaker.</returns>
+        public static PvSpeaker Create(int sampleRate, int bitsPerSample, int bufferSizeSecs = 20, int deviceIndex = -1)
+        {
+            return new PvSpeaker(sampleRate, bitsPerSample, bufferSizeSecs, deviceIndex);
+        }
```

Comment:
it doesn't look like we really need a Factory method here since it just forwards all args to the constructor as is. A factory is more useful if we're doing some operations with the input arguments before we call the constructor.

Reply:
I see - although, pvrecorder uses a factory method, I wonder if there's a reason there or if not, should we use the same pattern here for consistency?

Reply:
It looks like that one as well was not needed. None of the other SDKs use a static factory constructor in PvRecorder or PvSpeaker. I think we should break the pattern here and eventually remove the one from PvRecorder .NET. Sins of our fathers!